Jenkinsfile

```diff
@@ -0,0 +1,158 @@
+                                    sh script: 'vagrant ssh $CS -c \"sudo /home/vagrant/almalinux-deploy/almalinux-deploy.sh -d\"',
+                                        label: 'Start AlmaLinux Deploy'
+                                    sh script: 'vagrant destroy $CS -f',
+                                        label: 'Destroy the machine'
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        stage('Migrate legacy 8.5 stable systems to AlmaLinux 8') {
+            matrix {
+                axes {
+                    axis {
+                        name 'EL85'
+                        values 'centos8-5', 'oracle8-5', 'rhel8-5', 'rocky8-5'
+                    }
+                }
+                stages {
+                    stage('EL85 to AlmaLinux 8') {
+                        steps {
+                            retry(RETRY) {
+                                timeout(time: TIMEOUT, unit: 'MINUTES') {
+                                    sh script: 'vagrant up $EL85',
+                                        label: 'Create EL8.5 machine'
+                                    sh script: 'vagrant ssh $EL85 -c \"sudo /home/vagrant/almalinux-deploy/almalinux-deploy.sh\"',
+                                        label: 'Start AlmaLinux Deploy'
+                                    sh script: 'vagrant destroy $EL85 -f',
+                                        label: 'Destroy the machine'
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        stage('Migrate legacy 8.4 stable systems to AlmaLinux 8') {
+            matrix {
+                axes {
+                    axis {
+                        name 'EL84'
+                        values 'centos8-4', 'oracle8-4', 'rhel8-4', 'rocky8-4'
+                    }
+                }
+                stages {
+                    stage('EL8.4 to AlmaLinux 8') {
+                        steps {
+                            retry(RETRY) {
+                                timeout(time: TIMEOUT, unit: 'MINUTES') {
+                                    sh script: 'vagrant up $EL84',
+                                        label: 'Create EL8.4 machine'
+                                    sh script: 'vagrant ssh $EL84 -c \"sudo /home/vagrant/almalinux-deploy/almalinux-deploy.sh\"',
+                                        label: 'Start AlmaLinux Deploy'
+                                    sh script: 'vagrant destroy $EL84 -f',
+                                        label: 'Destroy the machine'
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+    post {
+        always {
+            sh script: 'vagrant destroy -f',
```

Comment:
Add slack notification of job run status

Reply:
I wanted to keep it minimal and universal. That's why it has the Source stage. In this way, the user can customize and add their preferred notification method by copy-pasting the pipeline file.